src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.Encrypt.cs

```diff
@@ -44,6 +45,10 @@ public sealed override byte[] Encrypt(CmsRecipientCollection recipients, Content
 
                 if (encodedContent.Length > 0)
                 {
+                    // Windows will throw if it encounters indefinite length encoding.
+                    // Let's reencode if that is the case
+                    ReencodeIfUsingIndefiniteLengthEncodingOnOuterStructure(ref encodedContent);
```

Comment:
Looks like this method isn't very hygenic; but we should fix it while we're here (and not make it worse).

The array from line 36 (id-data) is a copy, we should clear it and be nice to the user data.  If Reencode makes a copy then we need to clear that one, too.  But if we hit line 42 and don't rewrite it via reencode then we mustn't copy it, because the user still has access to that array.  (ContentInfo.Content returns the raw array, not a clone).